mypyc/irbuild/builder.py

```diff
@@ -240,6 +242,31 @@ def environment(self) -> Environment:
         return self.builder.environment
 
     # Helpers for IR building
+    def check_tagged_short_int(self, val: Value, line: int) -> Value:
+        int_tag = self.add(LoadInt(1, line, rtype=c_pyssize_t_rprimitive))
+        bitwise_and = self.binary_int_op(c_pyssize_t_rprimitive, val,
+                                         int_tag, BinaryIntOp.AND, line)
+        zero = self.add(LoadInt(0, line, rtype=c_pyssize_t_rprimitive))
+        check = self.binary_int_op(bool_rprimitive, bitwise_and, zero, BinaryIntOp.EQ, line)
+        return check
+
+    def binary_comparison_op(self, lhs: Value, rhs: Value, op: str, line: int) -> Value:
```

Comment:
Maybe rename this to `compare_tagged`? This is comparing tagged integers, specifically. Also I think that this should live in `ll_builder.py`, since this doesn't seem to depend on any high-level things such as AST nodes.

Add a docstring.